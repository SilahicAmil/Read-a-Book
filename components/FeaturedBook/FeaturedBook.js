import { Image, Pressable, StyleSheet, Text, View } from "react-native";

const FeaturedBook = ({ title, runtime, onPress }) => {
  return (
    <>
      <View style={styles.outerContainer}>
        <Text style={styles.featured}>Featured Book</Text>
        <Pressable
          style={({ pressed }) => (pressed ? styles.pressed : null)}
          onPress={onPress}
        >
          <Image
            source={{
              uri: "https://images.unsplash.com/photo-1529589941132-43606325dfb4?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1053&q=80",
            }}
            style={styles.image}
          />
          {/* Create this into custom component */}
          <View style={styles.descriptionContainer}>
            <Text style={styles.description}>{title}</Text>
          </View>
        </Pressable>
      </View>
    </>
  );
};

export default FeaturedBook;

const styles = StyleSheet.create({
  outerContainer: {
    width: "100%",
    padding: 8,
    height: 250,
    position: "relative",
    backgroundColor: "#FFF",
    justifyContent: "center",
    marginTop: 24,
    marginBottom: 32,
  },
  image: {
    width: "100%",
    height: "100%",
    borderRadius: 8,
  },
  featured: {
    fontSize: 24,
    marginBottom: 10,
    textAlign: "left",
    fontFamily: "DMSerif-Display",
  },
  descriptionContainer: {
    display: "flex",
    flexDirection: "row",
    width: "100%",

    backgroundColor: "#ccc",
    zIndex: 1,
    position: "absolute",
    opacity: 0.8,
    bottom: 0,
    padding: 8,
    borderBottomLeftRadius: 8,
    borderBottomRightRadius: 8,

    alignItems: "center",
    justifyContent: "space-between",
    shadowColor: "black",
    shadowOffset: {
      width: 0,
      height: 3,
    },
    shadowOpacity: 0.57,
    shadowRadius: 4.65,
    elevation: 8,
  },

  description: {
    fontFamily: "DMSerif-Display",
    fontWeight: "normal",
    fontSize: 24,
  },
  timeContainer: {
    justifyContent: "center",
    alignItems: "center",
    display: "flex",
    flexDirection: "row",
  },
  time: {
    fontFamily: "Kantumury-Pro",
    fontWeight: "normal",
    fontSize: 15,
    marginRight: 4,
  },
  listen: {
    marginLeft: 4,
    fontSize: 15,
    fontFamily: "Kantumury-Pro",
  },
  pressed: {
    opacity: 0.8,
  },
});
