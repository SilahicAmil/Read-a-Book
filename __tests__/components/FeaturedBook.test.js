import FeaturedBook from "../../components/FeaturedBook/FeaturedBook";
import renderer from "react-test-renderer";

describe("FeaturedBook Component", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<FeaturedBook />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
