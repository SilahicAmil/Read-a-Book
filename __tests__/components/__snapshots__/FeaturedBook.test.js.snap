// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedBook Component should render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFF",
      "height": 250,
      "justifyContent": "center",
      "marginBottom": 32,
      "marginTop": 24,
      "padding": 8,
      "position": "relative",
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "fontFamily": "DMSerif-Display",
        "fontSize": 24,
        "marginBottom": 10,
        "textAlign": "left",
      }
    }
  >
    Featured Book
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Image
      source={
        {
          "uri": "https://images.unsplash.com/photo-1529589941132-43606325dfb4?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1053&q=80",
        }
      }
      style={
        {
          "borderRadius": 8,
          "height": "100%",
          "width": "100%",
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ccc",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "bottom": 0,
          "display": "flex",
          "elevation": 8,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 0.8,
          "padding": 8,
          "position": "absolute",
          "shadowColor": "black",
          "shadowOffset": {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.57,
          "shadowRadius": 4.65,
          "width": "100%",
          "zIndex": 1,
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "DMSerif-Display",
            "fontSize": 24,
            "fontWeight": "normal",
          }
        }
      />
    </View>
  </View>
</View>
`;
